File: Safari-611.1.10/LayoutTests/http/tests/xmlhttprequest/resources/redirect-cors.php
<?php
    $url = $_GET['url'];
    $redirect_preflight = $_GET['redirect-preflight'];
    $access_control_allow_origin = $_GET['access-control-allow-origin'];
    $access_control_allow_credentials = $_GET['access-control-allow-credentials'];
    $access_control_allow_headers = $_GET['access-control-allow-headers'];

    if ($_SERVER['REQUEST_METHOD'] == "OPTIONS") {
        if ($redirect_preflight == "true") {
            header("HTTP/1.1 302");
            header("Location: $url");
        }
        else {
            header("HTTP/1.1 200");
        }
        header("Access-Control-Allow-Methods: GET");
        header("Access-Control-Max-Age: 1");
    } else if ($_SERVER['REQUEST_METHOD'] == "GET") {
        header("HTTP/1.1 302");
        header("Location: $url");
    }
    if (!is_null($access_control_allow_origin))
        header("Access-Control-Allow-Origin: $access_control_allow_origin");
    if (!is_null($access_control_allow_credentials))
        header("Access-Control-Allow-Credentials: $access_control_allow_credentials");
    if (!is_null($access_control_allow_headers))
        header("Access-Control-Allow-Headers: $access_control_allow_headers");
?>

File: LayoutTests/http/tests/xmlhttprequest/resources/redirect-cors.py
#!/usr/bin/env python3
import os
import sys
import time

from urllib.parse import parse_qs, unquote
query = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True)

sys.stdout.write('Content-Type: text/html\r\n')

url = query.get('url', [''])[0]

if os.environ.get('REQUEST_METHOD') == 'OPTIONS':
    if query.get('redirect-preflight', ['false'])[0] == 'true':
        sys.stdout.write(
            'status: 302\r\n'
            'Location: {}\r\n'.format(url)
        )
    else:
        sys.stdout.write('status: 200\r\n')
    sys.stdout.write(
        'Access-Control-Allow-Methods: GET\r\n'
        'Access-Control-Max-Age: 1\r\n'
    )

elif os.environ.get('REQUEST_METHOD') == 'GET':
    sys.stdout.write(
        'status: 302\r\n'
        'Location: {}\r\n'.format(url)
    )

allowOrigin = query.get('access-control-allow-origin', [None])[0]
allowCredentials = query.get('access-control-allow-credentials', [None])[0]
allowHeaders = query.get('access-control-allow-headers', [None])[0]

if allowOrigin is not None:
    sys.stdout.write('Access-Control-Allow-Origin: {}\r\n'.format(allowOrigin))
if allowCredentials is not None:
    sys.stdout.write('Access-Control-Allow-Credentials: {}\r\n'.format(allowCredentials))
if allowHeaders is not None:
    sys.stdout.write('Access-Control-Allow-Headers: {}\r\n'.format(allowHeaders))

sys.stdout.write('\r\n')
